compiler/rustc_middle/src/query/mod.rs

```diff
@@ -1719,6 +1719,12 @@ rustc_queries! {
         ensure_forwards_result_if_red
     }
 
+    query enforce_impl_non_lifetime_params_are_constrained(key: LocalDefId) -> Result<(), ErrorGuaranteed> {
+        desc { |tcx| "checking that `{}`'s generics are constrained by the impl header", tcx.def_path_str(key) }
+        cache_on_disk_if { true }
```

Comment:
It may be worth trying not to cache the value (unsure how we handle such query cache entries in general, en/decoding them is useless after all), but I think the regression is simply due to the extra nodes in the incremental graph.

Reply:
ok ill remove the last commit; i think we should just eat the perf

Reply:
see previous perf run for difference